.github/workflows/helm_pr_deployment.yml

```diff
@@ -0,0 +1,98 @@
+name: Deploy PR to Kubernetes
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  workflow_run:
+    workflows: ["tests"]
+    branches: [main]
+    types:
+      - completed
```

Comment:
Does this relate to the workflow defined in `.github/workflows/test.yml`? 
What's the reason for adding a workflow-trigger when your *only* job is conditional to the action having a `pull_request`?